### PR TITLE
feat(optimus): [RND-104008] Fix visual issues in dropdown button

### PR DESCRIPTION
**Pull request by witwash**
RND-104008

#### Summary

- Added animation for the chevron icon (duration and curve is matching the one on the web). 
- Code refactoring.

#### Testing steps

1. Open Storybook
2. Open Dropdown Button / Split Button story.
3. There should be an animation for the opening and closing (chevron rotation).

#### Follow-up issues

None.

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

**Review by kamil-grzadziela — APPROVED** [review]

File: optimus/lib/src/button/base_dropdown_button.dart [hunk 1]
```diff
@@ -157,10 +183,22 @@ class _BaseDropDownButtonState<T> extends State<BaseDropDownButton<T>>
...
 }
 
 const double _dropdownWidth = 280;
+
+extension on OptimusWidgetSize {
+  double get iconSize {
+    switch (this) {
+      case OptimusWidgetSize.small:
+        return 16;
+      case OptimusWidgetSize.medium:
```
Comment (kamil-grzadziela) [inline optimus/lib/src/button/base_dropdown_button.dart:199]: Why do we need medium if it's the same size as large?

Comment (witwash) [reply optimus/lib/src/button/base_dropdown_button.dart:199]: `OptimusWidgetSize` is widely used across the `Optimus` package. Design states that chevron size should be the same for the `OptimusWidgetSize.medium` and `OptimusWidgetSize.large`. 🎨

File: optimus/lib/src/button/base_dropdown_button.dart [hunk 2]
```diff
@@ -30,12 +30,35 @@ class BaseDropDownButton<T> extends StatefulWidget {
...
+      Tween<double>(begin: 0, end: 0.5);
+
   final _selectFieldKey = GlobalKey();
   bool _isHovering = false;
   bool _isTappedDown = false;
-  bool _isOpened = false;
+
   final _node = FocusNode();
+  late AnimationController _controller;
+  late Animation<double> _iconTurns;
```
Comment (ookami-kb) [inline optimus/lib/src/button/base_dropdown_button.dart:45]: Let's also make them final.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Review by ookami-kb — APPROVED** [review]

**Comment by ookami-kb** [comment]
@witwash the build is failing

**Comment by witwash** [comment]
@ookami-kb I'm on it ;)

**Pull request merged**